MediaBrowser.Api/PackageService.cs

```diff
@@ -188,13 +183,21 @@ public async Task<object> Get(GetPackages request)
         /// <exception cref="ResourceNotFoundException"></exception>
         public async Task Post(InstallPackage request)
         {
-            var package = string.IsNullOrEmpty(request.Version) ?
-                await _installationManager.GetLatestCompatibleVersion(request.Name, request.AssemblyGuid, typeof(PackageService).Assembly.GetName().Version, request.UpdateClass).ConfigureAwait(false) :
-                await _installationManager.GetPackage(request.Name, request.AssemblyGuid, request.UpdateClass, Version.Parse(request.Version)).ConfigureAwait(false);
+            var packages = await _installationManager.GetAvailablePackages().ConfigureAwait(false);
+            var package = _installationManager.GetCompatibleVersions(
+                    packages,
+                    request.Name,
+                    new Guid(request.AssemblyGuid),
+                    string.IsNullOrEmpty(request.Version) ? null : Version.Parse(request.Version),
+                    request.UpdateClass).FirstOrDefault();
```

Comment:
again, shouldn't we be taking maximum version, not just first in the list?

Reply:
Again, we are assuming the json is sorted (which it is atm, and should stay that way)